dit/utils.py

```diff
@@ -155,6 +159,9 @@ def apply_filter_from(data, date):
     for i in range(len(logbook) + 1):
         if i == len(logbook) or logbook[i]['in'] >= date:
             break
+        elif logbook[i]['out'] >= date:
+            logbook[i]['in'] = date
+            break
```

Comment:
Same as before, remove this break.

Also, the conditional on line 154 only needs to check the 'updated_at' key.